xdcs-frontend/src/app/view/task-creation/task-creation.component.ts

```diff
@@ -0,0 +1,171 @@
+            type: entry.type,
+          },
+          hasChildren: entry.type === TreeFileType.DIRECTORY,
+        };
+      }));
+  }
+
+  private loadRoot() {
+    this.loadChildren(null).then(nodes => this.files = nodes);
+  }
+
+  private loadFile(path: string): Promise<TreeDirectory> {
+    function mapType(type: 'regular' | 'directory' | 'link'): TreeFileType {
+      switch (type) {
+        case 'regular':
+        default:
+          return TreeFileType.FILE_OTHER;
+        case 'directory':
+          return TreeFileType.DIRECTORY;
+        case 'link':
+          return TreeFileType.FILE_SYMLINK;
+      }
+    }
+
+    console.log(this.taskDefinitionsService.getTaskDefinitionWorkspaceFile({
+      taskDefinitionId: this.taskId, path,
+    }));
+
+    return this.taskDefinitionsService.getTaskDefinitionWorkspaceFile({
+      taskDefinitionId: this.taskId, path,
+    }).pipe(map(fileDto => ({
+      entries: fileDto.children.map(child => ({
+        name: child.name,
+        type: mapType(child.type),
+      })),
+    }))).toPromise();
+  }
+
+  private nodeToPath(node: TreeNode): string {
+    if (!node || !node.data.name) {
+      return '/';
+    }
+
+    const name = node.data.name;
+    if (node.parent) {
+      const parentPath = this.nodeToPath(node.parent);
+      const parentPathWithSlash = parentPath.endsWith('/') ? parentPath : parentPath + '/';
+      return parentPathWithSlash + name;
+    } else {
+      return '/' + name;
+    }
+  }
+
+  loadTask(taskId): void {
+    this.taskId = taskId;
+    this.taskDefinitionsService.getTaskDefinition({
+      taskDefinitionId: this.taskId,
+    }).pipe(first()).subscribe(def => {
+      this.taskDefinition = def;
+    });
+    this.nodesService.getNodes()
+      .pipe(first())
+      .subscribe((nodes: NodesDto) => {
+        this.nodes = nodes.items;
+      });
+  }
+
+  isDefinitionConfigured(): boolean {
+    const config = this.taskDefinition.config;
+    if (config !== undefined) {
+      if (['cuda', 'opencl'].includes(config.type)) {
+        if (config.kernelfile !== null && config.kernelname !== null) {
+          return true;
+        }
+      } else if (config.type === 'docker') {
+        if (config.dockerfile !== undefined) {
+          return true;
+        }
+      } else if (config.type === 'script') {
+        if (config.scriptfile !== undefined) {
+          return true;
+        }
+      }
+    }
+    return false;
+  }
+
+  addDevice() {
+    if (!this.usedNodes) {
+      this.usedNodes = [];
+    }
+    const param: NodeDto = {};
+    this.usedNodes.push(param);
+  }
+
+  removeDevice(param: KernelParamDto) {
+    const nodes = this.usedNodes;
+    this.usedNodes = nodes.filter(item => item !== param);
+  }
+
+  moveDeviceUp(param: KernelParamDto) {
+    const index = this.usedNodes.indexOf(param);
+    if (index > 0 && index < this.usedNodes.length) {
+      this.swapDevices(index - 1, index);
+    }
+  }
+
+  moveDeviceDown(param: KernelParamDto) {
```

Comment:
przecież to jest copy-paste z task-def-config

Reply:
zrobiony osobny component